packages/gatsby/src/utils/worker/__tests__/schema.ts

```diff
@@ -0,0 +1,147 @@
+import * as path from "path"
+import fs from "fs-extra"
+import { DocumentNode } from "graphql"
+import { build } from "../../../schema"
+import sourceNodesAndRemoveStaleNodes from "../../source-nodes"
+import { saveStateForWorkers, store } from "../../../redux"
+import { loadConfigAndPlugins } from "../../../bootstrap/load-config-and-plugins"
+import {
+  createTestWorker,
+  describeWhenLMDB,
+  GatsbyTestWorkerPool,
+} from "./test-helpers"
+import { getDataStore } from "../../../datastore"
+import { CombinedState } from "redux"
+import { IGatsbyState } from "../../../redux/types"
+import type { watch as ChokidarWatchType } from "chokidar"
+
+let worker: GatsbyTestWorkerPool | undefined
+
+// when we load config and run sourceNodes on "main process" we start file watchers
+// because of default `gatsby-plugin-page-creator` which would prevent test process from
+// exiting gracefully without forcing exit
+// to prevent that we keep track of created watchers and close them after all tests are done
+const mockWatchersToClose = new Set<ReturnType<typeof ChokidarWatchType>>()
+jest.mock(`chokidar`, () => {
+  const chokidar = jest.requireActual(`chokidar`)
+  const originalChokidarWatch = chokidar.watch
+
+  chokidar.watch = (
+    ...args: Parameters<typeof ChokidarWatchType>
+  ): ReturnType<typeof ChokidarWatchType> => {
+    const watcher = originalChokidarWatch.call(chokidar, ...args)
+    mockWatchersToClose.add(watcher)
+    return watcher
+  }
+
+  return chokidar
+})
+
+describeWhenLMDB(`worker (schema)`, () => {
+  let state: CombinedState<IGatsbyState>
```

Comment:
Doesn't matter functionally, but what do you think about using name for it that is a bit more descriptive that this is not local state, but state from worker? I was reading tests and did have to scroll up to make sure that `state` is something that came from worker and not from within same process. Something like `stateFromWorker` (or maybe something else to add a bit of contextual info about origin of it)